app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb

```diff
@@ -65,7 +65,18 @@ def images
 
   def key_pairs
     return @key_pairs if @key_pairs.any?
-    @key_pairs = compute_service.handled_list(:key_pairs, {}, openstack_admin?)
+    users = User.all
+
+    results = users.map do |u|
+      next if u.settings[:openstack_user_id].nil?
+      @key_pairs = compute_service.handled_list(:key_pairs, {:user_id => u.settings[:openstack_user_id]}, openstack_admin?)
+
+      @key_pairs.map do |kp|
+        kp.user_id = u.settings[:openstack_user_id]
```

Comment:
@andyvesel what does `u.settings[:openstack_user_id]` look like?  It probably isn't equal to our `user.id` in our database right?